src/controllers/softtakeover.cpp

```diff
@@ -102,32 +102,74 @@ bool SoftTakeover::ignore(ControlObject* control, double newParameter) {
 
     qint64 currentTime = Time::elapsedMsecs();
     // We will get a sudden jump if we don't ignore the first value.
-    if (m_time == 0) {
+    if (m_time == -1) {
         ignore = true;
         // Change the stored time (but keep it far away from the current time)
         //  so this block doesn't run again.
-        m_time = 1;
-//         qDebug() << "SoftTakeover::ignore: ignoring the first value"
-//                  << newParameter;
-    } else if ((currentTime - m_time) > SUBSEQUENT_VALUE_OVERRIDE_TIME_MILLIS) {
-        // don't ignore value if a previous one was not ignored in time
+        m_time = -2;    // Tests like to use 0
+        //qDebug() << "ignoring the first value" << newParameter;
+    } else {
         const double currentParameter = control->getParameter();
         const double difference = currentParameter - newParameter;
-        const double prevDiff = currentParameter - m_prevParameter;
-        if ((prevDiff < 0 && difference < 0) ||
-                (prevDiff > 0 && difference > 0)) {
-            // On same side of the current parameter value
-            if (fabs(difference) > m_dThreshold && fabs(prevDiff) > m_dThreshold) {
-                // differences are above threshold
-                ignore = true;
-//                 qDebug() << "SoftTakeover::ignore: ignoring, not near"
-//                          << newParameter << m_prevParameter << currentParameter;
+        if (m_prevParameter != -1 && fabs(difference) > m_dThreshold) {
+            // New parameter is far away from current
+            const double prevDiff = currentParameter - m_prevParameter;
+            if (prevDiff * difference <= 0) {
```

Comment:
Ah, nevermind. the condition is catched below. 

You can use 
`if (prevDiff * difference < 0) {`
here